File: trip-kit/src/test/java/com/skedgo/android/tripkit/ConfigsTest.java
package com.skedgo.android.tripkit;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(TestRunner.class)
@Config(constants = BuildConfig.class)
public class ConfigsTest {
  /**
   * Only regionEligibility and context are mandatory.
   * Others are optional, and debuggable is false by default.
   */
  @Test public void build() {
    final Configs configs = Configs.builder()
        .regionEligibility("")
        .context(RuntimeEnvironment.application)
        .build();
    assertThat(configs.debuggable()).isFalse();
    assertThat(configs.isUuidOptedOut()).isFalse();
  }

  @Test(expected = NullPointerException.class)
  public void regionEligibilityIsNonNull() {
    Configs.builder()
        .regionEligibility(null)
        .context(RuntimeEnvironment.application)
        .build();
  }

  @Test(expected = NullPointerException.class)
  public void contextIsNonNull() {
    Configs.builder()
        .regionEligibility("")
        .context(null)
        .build();
  }
}

File: trip-kit/src/test/java/com/skedgo/android/tripkit/ConfigsTest.kt
package com.skedgo.android.tripkit

import com.nhaarman.mockito_kotlin.mock
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.robolectric.RuntimeEnvironment
import skedgo.tripkit.configuration.ApiKey

class ConfigsTest {
  @Test fun uuidOptedOutShouldBeFalseByDefault() {
    val configs = Configs.builder()
        .apiKey { ApiKey("Doesn't look like anything to me") }
        .context(mock())
        .build()
    assertThat(configs.isUuidOptedOut()).isFalse()
  }

  @Test fun shouldNotBeDebuggableByDefault() {
    val configs = Configs.builder()
        .apiKey { ApiKey("Doesn't look like anything to me") }
        .context(mock())
        .build()
    assertThat(configs.debuggable()).isFalse()
  }

  @Test(expected = NullPointerException::class)
  fun regionEligibilityShouldBeNonNull() {
    Configs.builder()
        .apiKey(null)
        .context(RuntimeEnvironment.application)
        .build()
  }

  @Test(expected = NullPointerException::class)
  fun contextShouldBeNonNull() {
    Configs.builder()
        .apiKey { ApiKey("Doesn't look like anything to me") }
        .context(null)
        .build()
  }
}
